src/api/Metadata.js

```diff
@@ -756,23 +758,38 @@ class Metadata extends File {
             errorCallback(getBadPermissionsError(), this.errorCode);
             return;
         }
-
         this.successCallback = successCallback;
         this.errorCallback = errorCallback;
 
+        const convertFieldsToKeyValueObject = (template.fields || []).reduce((acc, obj) => {
+            if (obj.value !== undefined) {
+                acc[obj.key] = obj.value;
+            }
+            return acc;
+        }, {});
+
```

Comment:
Can we use `fromEntries` here?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries

```suggestion
        const convertFieldsToKeyValueObject = Object.fromEntries(template.fields || []);
```

Reply:
I had to change it back to `reduce` because `fromEntries` didn't pass the `flow check`.